backend/fms_core/template_importer/importers/sample_pooling_planning.py

```diff
@@ -0,0 +1,286 @@
+
+    def initialize_data_for_template(self):
+        pass
+
+    def import_template_inner(self):
+        sheet = self.sheets["SamplesToPool"]
+
+        MAX_SOURCE_CONTAINERS = 8
+        MAX_DESTINATION_CONTAINERS = 24
+        TUBE = "tube"
+
+        source_containers_barcodes = set()
+        destination_containers_barcodes = set()
+
+        pooling_mapping_rows = []
+        container_unique_id = get_unique_id()
+        base_error_rows = []
+        # For each row initialize the object that is going to be prefilled in the pooling template
+        for row_id, row_data in enumerate(sheet.rows):
+            pooling_type = str_cast_and_normalize(row_data["Type"])
+            source_sample = {
+                "name": str_cast_and_normalize(row_data["Source Sample Name"]),
+                "container": {"barcode": str_cast_and_normalize(row_data["Source Container Barcode"])},
+                "coordinates": str_cast_and_normalize(row_data["Source Container Coord"]),
+                "depleted": str_cast_and_normalize(row_data["Source Depleted"]),
+            }
+
+            pool_name = str_cast_and_normalize(row_data["Pool Name"]) if str_cast_and_normalize(row_data["Pool Name"]) else ""
+
+            pool = {
+                "name": pool_name,
+                "coordinates": None, # No coordinate in tube
+                "container": {
+                    "barcode": pool_name + "_" + str(container_unique_id), # Artificial barcode built from the pool name and a sequential number
+                    "name": pool_name + "_" + str(container_unique_id), # Same as barcode
+                    "kind": TUBE, # tube
+                    "coordinates": None, # Do not place into a parent
+                    "parent_barcode": None, # Do not place into a parent
+                },
+            }
+
+            measurements = {
+                'na_quantity': float_to_decimal_and_none(row_data['NA Quantity Used (ng)']),
+            }
+
+            source_containers_barcodes.add(source_sample["container"]["barcode"])
+            destination_containers_barcodes.add(pool["container"]["barcode"])
+
+            pooling_kwargs = dict(
+                type=pooling_type,
+                source_sample=source_sample,
+                pool=pool,
+                measurements=measurements,
+            )
+
+            (result, pooling_row_mapping) = self.handle_row(
+                row_handler_class=SamplePoolingPlanningRowHandler,
+                sheet=sheet,
+                row_i=row_id,
+                **pooling_kwargs
+            )
+
+            if (pooling_row_mapping is None):
+                base_error_rows.append(sheet.rows_results[row_id]["row_repr"])
+            
+            pooling_mapping_rows.append(pooling_row_mapping)
+
+        if (len(source_containers_barcodes) > MAX_SOURCE_CONTAINERS):
+            self.base_errors.append(f"Too many source containers ({MAX_SOURCE_CONTAINERS}) for the robot.")
```

Comment:
Wouldn't it be better to test the total `source_containers_barcodes` length before row handling? Well that might lead to more codes, so it's a trade-off.